classes/class-p4-master-site.php

```diff
@@ -674,9 +674,14 @@ public function p4_auto_generate_excerpt( $post_id, $post ) {
 			remove_action( 'save_post', [ $this, 'p4_auto_generate_excerpt' ], 10 );
 
 			// Generate excerpt text.
-			$post_excerpt   = strip_shortcodes( $post->post_content );
-			$post_excerpt   = apply_filters( 'the_content', $post_excerpt );
-			$post_excerpt   = str_replace( ']]>', ']]&gt;', $post_excerpt );
+			preg_match( "'<p>(.*?)</p>'", $post->post_content, $match_paragraph );
+			if ( count( $match_paragraph ) > 0 ) {
+				$post_excerpt = $match_paragraph[1];
```

Comment:
This will match only the first paragraph, which will work in most cases but maybe there's cases where the first paragraph is only a couple of words and we want to include more in the excerpt?

Reply:
I agree on @Inwerpsel point, in that case the except would be very small.